### ISSUE TITLE
Usage Xcode 9's recommended project settings causes warnings

### ISSUE THREAD

**Issue by weyert** [Instagram/IGListKit#955]
## New issue checklist

- [X] I have reviewed the [`README`](https://github.com/Instagram/IGListKit/blob/master/README.md) and [documentation](http://instagram.github.io/IGListKit)
- [X] I have searched [existing issues](https://github.com/Instagram/IGListKit/issues) and this is not a duplicate

### General information

- `IGListKit` version: `master`
- iOS version(s): N/A
- CocoaPods/Carthage version: 1.3.1
- Xcode version: 9.0
- Devices/Simulators affected: N/A
- Reproducible in the demo project? (Yes/No): Yes
- Related issues: 

### Debug information

When you start using Xcode 9 for a project and use `IGListKit` through Cocoapods Xcode suggests to use it's recommended settings for the Pods and your own project. if you apply these recommended settings. A few compiler warnings occur in the codebase. Especially, around the ListAdapter code:

```
/Pods/IGListKit/Source/Internal/IGListBatchUpdates.h:26:64: This block declaration is not a prototype
```

**Comment by weyert** [comment]
Already fixed in an existing PR #942. Closing ticket

**Comment by jazz-mobility** [comment]
These changes are not available in 3.1.1, when are you planning to release them?

**Comment by rnystrom** [comment]
@jazz-mobility wrapping up 3.2 now!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
